collector/helper.go

```diff
@@ -16,3 +20,33 @@ func newUpMetric(namespace string) prometheus.Gauge {
 		Help:      "Status of the last metric scrape",
 	})
 }
+
+// MergeLabels merges two maps of labels.
+func MergeLabels(a map[string]string, b map[string]string) map[string]string {
+	c := make(map[string]string)
+
+	for k, v := range a {
+		c[k] = v
+	}
+	for k, v := range b {
+		c[k] = v
+	}
+
+	return c
+}
+
+// reservedLabelPrefix is a prefix which is not legal in user-supplied label names.
+const reservedLabelPrefix = "__"
+
+// IsValidLabelName does equivalent validation to checkLabelName in prometheus/client_golang
+func IsValidLabelName(ln string) bool {
```

Comment:
could we reuse validation methods from https://godoc.org/github.com/prometheus/common/model? We already vendor that dependency https://github.com/nginxinc/nginx-prometheus-exporter/blob/master/vendor/github.com/prometheus/common/model/labels.go

Also note that you don't validate the label value at all.